tests/test_Dataset.py

```diff
@@ -643,6 +643,27 @@ def _test(sizes: List[int], partition_epoch: int, expected: List[List[int]]):
     _test([5, 5] + [10] * 7, 5, [[5, 5, 10], [10, 10], [10, 10], [10], [10]])
 
 
+NestedSize = Union[int, List["NestedSize"]]
+
+
+def test_ConcatFilesDataset_get_files_per_sub_epochs_nested():
```

Comment:
I don't understand this test. `_get_files_per_sub_epochs` is not really about the nested files at all.

Reply:
I don't think we need to have this test case. If you want to have a test case for the nested aspect, this would not be about `_get_files_per_sub_epochs`.